pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -81,12 +89,7 @@ stdenv.mkDerivation {
     # official version constraints (as recorded in default.nix). In some cases
     # you _may_ be able to smudge version constraints, just know that you're
     # embarking into unknown and unsupported territory when doing so.
-    broken = let cudaVer = lib.getVersion cudatoolkit; in
-      !(
-        lib.versionAtLeast cudaVer minCudaVersion
-        && lib.versionAtLeast maxCudaVersion cudaVer
-      );
-
+    broken = !(elem cudatoolkit.majorMinorVersion supportedCudaVersions);
```

Comment:
There are some cuDNN versions that have patch x.y.z constraints instead of simply x.y constraints. How will we handle those?